### ISSUE TITLE
pccnn的网络权重下载，对于不同的b值、不同的输入扩散方向数，您公开的权重怎么获取

### ISSUE THREAD

**Issue by Sxwang0320** [m-lyon/dmri-pcconv#2]
亲爱的教授：
我在复现您的实验时，想要直接使用您训练好的权重进行预测，但是我发现每个pccnn网络只有一个可下载的权重，其他的不同输入方向数和不同b值的权重，您可以共享一下吗，谢谢

**Comment by m-lyon** [comment]
Each PCCNN model was trained with data that had a variety of subsampling ratios and b-values. Specifically, the number of input volumes varied from between 6-20, for b=1000, b=2000, and b=3000. Therefore each model should be capable of predicting any data from within those ranges and b-values.

An important aside: each training example had exactly 1 b-value as _input_ and 1 b-value as _output_. So, for example a valid input would be 6-20 b=1000 volumes, with output 10 b=2000 volumes. All of this is to say, if you were to use multiple b-values in one _input_, for example 6 b=1000 volumes and 6 b=2000 volumes, then this would fall **outside of the training data**.

**Comment by Sxwang0320** [comment]
教授您好：
我想向您请教的就是您说的这些不同子采样率训练好的，每个模型的权重，因为您是在4张A100上训练了200，000次迭代，我需要您的帮助，利用您训练好的权重来预测结果，只用于测试。

**Comment by m-lyon** [comment]
You should be able to use one set of weights for different subsampling rates. For example if you wanted to use a subsampling rate of 12, you just need to ensure that the `bvec_in` and `dmri_in` arguments to `PCCNNBvPredictionProcessor.run_subject` have 12 b-vectors and volumes respectively.

**Comment by Sxwang0320** [comment]
这样我确实可以利用您提供的权重（一组）计算不同的输入方向数的预测结果，但是这样能保证您的方法的性能吗，针对不同的bvec_in，您的方法是不需要在重新训练了吗？

**Comment by m-lyon** [comment]
So long as the `bvec_in` number is between 6 and 20 you should not need to retrain the model, as this range is within the training data regime. One advantage of training this way is that the model can be used for a range of `bvec_in` sampling rates without having to retrain the model.

**Comment by Sxwang0320** [comment]
非常感谢您，另外我如果想要用b值为2000的数据作为输入，预测b值为2000的输出同样使用您提供的下载权重来预测吗

**Comment by m-lyon** [comment]
Yes, during training all pairwise combinations of b-values (from the set of 1000, 2000, and 3000) were used. This includes same-shell (i.e. b=2000 to b=2000) and different-shell (i.e. b=2000 to b=3000).
